src/Admin/Extension/PostalCodeFilterAdminExtension.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+namespace AppBundle\Admin\Extension;
+
+use Sonata\AdminBundle\Admin\AbstractAdminExtension;
+use Sonata\AdminBundle\Datagrid\DatagridMapper;
+use Sonata\DoctrineORMAdminBundle\Filter\CallbackFilter;
+use Sonata\DoctrineORMAdminBundle\Datagrid\ProxyQuery;
+use Symfony\Component\Form\Extension\Core\Type\TextType;
+
+class PostalCodeFilterAdminExtension extends AbstractAdminExtension
+{
+    public function configureDatagridFilters(DatagridMapper $datagridMapper)
+    {
+        $datagridMapper
+            ->add('postalCode', CallbackFilter::class, [
+                'label' => 'Code postal (préfixe)',
+                'show_filter' => true,
+                'field_type' => TextType::class,
+                'callback' => function (ProxyQuery $qb, string $alias, string $field, array $value) {
+                    if (!$value['value']) {
+                        return;
```

Comment:
`return false;`
This indicates whether the filter is applied on this field or not